modules/covid_vaccine/app/jobs/covid_vaccine/enrollment_upload_job.rb

```diff
@@ -12,14 +12,34 @@ class EnrollmentUploadJob
 
     def perform(batch_id)
       Rails.logger.info('Covid_Vaccine Enrollment_Upload: Start', batch_id: batch_id)
+      resolve_facilities(batch_id)
+      upload(batch_id)
+      StatsD.increment(STATSD_SUCCESS_NAME)
+    rescue => e
+      handle_errors(e, batch_id)
+    end
+
+    private
+
+    def resolve_facilities(batch_id)
+      update_count = 0
+      CovidVaccine::V0::ExpandedRegistrationSubmission.where(batch_id: batch_id).find_each do |submission|
+        resolver = CovidVaccine::V0::FacilityResolver.new
+        mapped_facility = resolver.resolve(submission)
+        submission.eligibility_info = { preferred_facility: mapped_facility }
```

Comment:
Is state being checked prior to this job running?  If not, this might overwrite some existing eligibility_info. After the MPI lookup succeeds the ICN is saved to eligibility_info.  if state == `enrollment_complete` or state == `registered` there should be an ICN saved in eligibility_info

Reply:
@stephenBarrs ah ok, thanks for pointing that out, I updated it to check if `submission.eligibility_info` exists, and if it does merge it with the existing data, and if it does not then set it to `{ preferred_facility: mapped_facility }`